lib/reply.js

```diff
@@ -666,18 +667,20 @@ function sendStream (payload, res, reply) {
   } else {
     reply.log.warn('response will send, but you shouldn\'t use res.writeHead in stream mode')
   }
-  payload.pipe(res)
+
+  sendStreamTrailer(payload, res, reply)
+
+  payload.pipe(res, { end: reply[kReplyTrailers] === null })
 }
 
 function sendTrailer (payload, res, reply) {
   if (reply[kReplyTrailers] === null) return
-  const trailerHeaders = Object.keys(reply[kReplyTrailers])
   const trailers = {}
   let handled = 0
-  for (const trailerName of trailerHeaders) {
-    if (typeof reply[kReplyTrailers][trailerName] !== 'function') continue
-    handled--
 
+  const _trailers = Object.entries(reply[kReplyTrailers]).filter(([k, v]) => typeof v === 'function')
```

Comment:
is there good choice without using `.filter`?
The previous implementation using a single `variables` will count wrong and have racing condition.

cc @Uzlopak maybe you will have some idea?